doc/source/zuul.rst

```diff
@@ -65,5 +66,354 @@ each commit is correctly tested.
+  with no direct access using public IP addresses.
+
+* Zookeeper instance used by Zuul is a dedicated instance with no external access.
+
+* SQL DB used by Zuul is a dedicated instance with no public IP address.
+
+* API and SSH access to git hosting can be additionally protected by the
+  whitelisting of Zuul external IP address.
+
+Backup and Restore
+~~~~~~~~~~~~~~~~~~
+
+Zuul is build on the principles of storing all required information in git.
+This is applicable for the configuration of which jobs are executed for which
+project, as well as what is the Zuul configuration. This makes Backup more or
+less obsolete. Of course there are some parts of the installation that
+require backups:
+
+* private/public keys for the project secrets (private keys are in addition protected by password).
+
+Details on the methods can be found `here <https://zuul-ci.org/docs/zuul/reference/client.html>`_.
+
+Certificate Handling
+~~~~~~~~~~~~~~~~~~~~~
+
+There are few types of certificates used in Zuul:
+
+* Zookeeper client TLS certificates
+* TLS certificates for the API/UI (Web access)
+* API keys and private certificates for SSH and API access to git hoster.
+
+Those certificates must be maintained according to the security
+requirements and deployment specifics. In general it is preferred to use
+short living self-signed certificates for the Zookeeper cluster as well as
+LetsEncrypt certificates for Web access.
+
+User and account Management
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generally Zuul does not support user accounts. It mainly communicates with
+git hosting systems with appropriate credentials and has no information about
+particular users proposing changes there.
+
+Zuul supports optional `Tenant Scoped REST API
+<https://zuul-ci.org/docs/zuul/discussion/tenant-scoped-rest-api.html>`_, but
+this is currently not enabled in the current installation.
+
+Operational accounts
+^^^^^^^^^^^^^^^^^^^^
+
+There are not granular operator accounts in Zuul installation. There is only one account allowing operate the system.
+
+Technical and M2M accounts
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Every component of Zuul only communicates to Zookeeper. For this Zookeeper
+client TLS certificate is used. No other technical or M2M accounts exist on
+the system.
+
+Communication Matrix (internal)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As mentioned above Zuul components communicate with each other only through
+Zookeeper. In order to keep communication simple those missing connections are
+not mentioned explicitly.
+
+.. list-table::
+
+   * - From \\ To
+     - zookeeper
+     - vault
+   * - nodepool-builder
+     - TLS(2281)
+     - TLS(8200)
+   * - nodepool-launcher
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-web
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-merger
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-executor
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-scheduler
+     - TLS(2281)
+     - TLS(8200)
+   * - zookeeper
+     - TLS(2888,3888)
```

Comment:
Is it zookeper to zookeper communication through ports 2888 and 3888 ? This is not visible in any diagram.